### PR TITLE
Add Eslint and prettier

### LINKED ISSUE #63: Add linting

### LINKED ISSUE #87: Set up auto-formatting
We have a nice, consistent pattern across our other repos of setting up prettier.js to format according to some rules. (The rules can almost be arbitrary -- it doesn't matter).

Not having prettier auto-format on save means more cognitive overhead when developing, and more noise in change sets.

### PR DESCRIPTION
**Pull request by lalexgap**
I was running into some format issues (since I have prettier running locally) so I bit the bullet and setup eslint and prettier.

Fixes #87
Fixes #63

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/.yarn/*
```
Comment (kerzhner) [inline .eslintignore:1]: Why is there a `/*` at the end of this path but not others?

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+/artifacts
+/cache
+node_modules
```
Comment (kerzhner) [inline .eslintignore:4]: Why are there `/` at the start of other paths, but not here?

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+/contract-types
+.vscode
```
Comment (kerzhner) [inline .eslintignore:6]: Why are there `/` at the start of other paths, but not here?

File: .prettierrc.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (kerzhner) [inline .prettierrc.json:1]: Is this empty config needed?

**Review by kerzhner — APPROVED** [review]

File: .eslintignore (hunk 1 above)
Comment (lalexgap) [reply .eslintignore:1]: Not sure, this is mostly a copy and paste of our `.gitignore`

File: .prettierrc.json (hunk 4 above)
Comment (lalexgap) [reply .prettierrc.json:1]: I think so?

I just followed the simple [instructions](https://prettier.io/docs/en/install.html) for installing prettier which included creating an empty config file.

File: .eslintignore (hunk 3 above)
Comment (lalexgap) [reply .eslintignore]: c0ffac6

File: .eslintignore (hunk 2 above)
Comment (lalexgap) [reply .eslintignore]: Resolved in c0ffac6

File: .eslintignore (hunk 1 above)
Comment (lalexgap) [reply .eslintignore]: Resolved in c0ffac6

**Pull request merged**